### DISCUSSION TITLE
Git flow, changelog, projects and workspace/monorepo

### DISCUSSION THREAD

**Discussion by v41e** [projen/projen#2782] [Q&A]
Hi Projen community,

Is there a way to work with gitflow (eg feature/*, develop, release/*, main, hotflix/*) ?

Also, how the changelog is generated ? I use https://github.com/git-chglog/git-chglog on my projects, it's really handy and allow high customization.

Also I would like to know if projen should be use only on TS project, or could it be use in any languages (go, rust, dart...) ?

And last question there is a parent / subprojects logic but can it helps to build pnpm workspace or nx monorepo for example?

Cheers,

**Comment by v41e** [comment]
Another question, why there is no cdktf app project by default?

**Reply by mrgrain** [reply]
Just hasn't been created yet by the community

**Comment by mrgrain** [comment]
> Is there a way to work with gitflow (eg feature/, develop, release/, main, hotflix/*) ?

projen doesn't really prescribe your branching model. However by default it assumes a single default branch (`main`) for everything, which might not be what you need with gitflow, mostly because `develop` and `release`

> Also, how the changelog is generated ? I use https://github.com/git-chglog/git-chglog on my projects, it's really handy and allow high customization.

It currently uses `standard-version` which uses `conventional-changelog`. There's no way to customize this without re-implementing the Release Component (but you can do that).

> Also I would like to know if projen should be use only on TS project, or could it be use in any languages (go, rust, dart...) ?

You can use projen for anything you like. We have built-in support for JS, TS, Python, Java.


> And last question there is a parent / subprojects logic but can it helps to build pnpm workspace or nx monorepo for example?

It does help, but it is a very low-level feature. Plugins implement more complex workflows e.g. for `nx`:
https://www.npmjs.com/package/@aws-prototyping-sdk/nx-monorepo

**Reply by v41e** [reply]
Hi @mrgrain,

Thanks a lot for your responses!

I used standard-version and switched to git-chglog years ago due to the time needed to create the changelog in a large repository with a large number of commits (it was very long in standard-version but maybe they made some improvements since), use of sandard-version is acceptable to me the changelog is good, no need to customize.

I would really like to use projen but I want to keep gitflow, how can I deal with it, maybe I can run `npx projen release` on the `release/*` branch in order to increment the minor number or the `hotfix/*` branch with only fix to increment the patch? Am I right?

I'll have a look at nx-monorepo thanks a lot!
